tools/make/diplomat-coverage/src/allowlist.rs

```diff
@@ -360,6 +371,13 @@ lazy_static::lazy_static! {
         "icu::calendar::Date::into_atomic_ref_counted",
         "icu::calendar::Date::as_borrowed",
 
+        // Unchecked formatting: FFI does not have the type system safety. All format
+        // methods are already "unchecked" in the sense that anything goes.
+        "icu::datetime::DateTimeInputUnchecked",
```

Comment:
suggestion (nb): usually when two Rust APIs correspond to one C API, I err on the side of adding hidden rust_links to the same API. e.g. see how GraphemeClusterSegmenterUtf8 links to PotentiallyIllFormedUtf8 and Utf8.

My tendency is to try and keep this file small if possible. In the medium term I'd like rustlink to have a better way of specifying such links in bulk.

Reply:
The comment wasn't really correct; I rewrote it and moved it to the "not for 2.0" section.